TheDirectoryApp/TheDirectoryApp/ViewControllers/ListViewController.swift

```
@@ -0,0 +1,359 @@
+        
+        loadList()
+}
+    func setUpActivityView()
+    {
+        activityIndicator = UIActivityIndicatorView(style: UIActivityIndicatorView.Style.large)
+        activityIndicator.center = view.center
+        self.view.addSubview(activityIndicator)
+        activityIndicator.startAnimating()
+    }
+    override func viewWillAppear(_ animated: Bool) {
+        self.bookedRoomsButton.titleLabel?.textColor = UIColor(red: 196.0/255.0, green: 2.0/255.0, blue: 2.0/255.0, alpha: 1)
+    }
+
+    override func viewDidAppear(_ animated: Bool) {
+        
+        // handling scenarios of app in background and foreground
+        let foregroundNotificationCenter = NotificationCenter.default
+        foregroundNotificationCenter.addObserver(self, selector:#selector(handleUIInBakgroundModes),name: UIApplication.willEnterForegroundNotification, object: nil)
+        //
+        
+        if categorySelected == ListViewController.MEETING_ROOMS
+        {
+        self.title = "Meeting Rooms"
+        }
+        else if categorySelected == ListViewController.PEOPLE
+        {
+            self.title = "Emplyoees"
+        }
+        self.bookedRoomsButton.titleLabel?.textColor = UIColor(red: 196.0/255.0, green: 2.0/255.0, blue: 2.0/255.0, alpha: 1)
+        updateUIForAvailableRooms()
+    }
+    @objc func handleUIInBakgroundModes()
+    {
+        updateUIForAvailableRooms()
+    }
+    private func loadList()
+    {
+        activityIndicator.startAnimating()
+        if categorySelected == ListViewController.MEETING_ROOMS
+        {
+            bookedRoomsButton.isHidden = false
+            availableRoomsButton.isHidden = false
+            viewModel.fetchMeetingRoomsDetails(completion: { list, error in
+                
+                    if let error = error {
+                        print("Error : \(error.localizedDescription)")
+                        return
+                    }
+                    guard let listOfRooms = list  else { return }
```

Have you noticed that in this `guard`, if the list is nil, then the spinner will never stop and will fail silently?